api/src/paths/dwc/submission/{datasetId}/artifacts.ts

```diff
@@ -134,12 +152,27 @@ export function getArtifactsByDatasetId(): RequestHandler {
       const artifacts = await artifactService.getArtifactsByDatasetId(datasetId);
 
       const artifactsWithRules = await Promise.all(
-        artifacts.map(async (artifact) => {
-          const appliedSecurity = await securityService.getSecurityAppliedStatus(artifact.artifact_id);
-          return {
-            ...artifact,
-            supplementaryData: { persecutionAndHarm: appliedSecurity }
-          };
+        artifacts.map((artifact) => {
```

Comment:
this seems like a lot of logic for an endpoint to have, might be better wrapped in a new service function

Reply:
I decided to tackle this code in the endpoint as to pollute the Artifacts Service with one-off methods that are only used by one endpoint. To that end, I especially favor this because 1) endpoints will cobble together supplementary data in different ways, 2) the code doesn't have any side effects (it doesn't modify any records in the database), and 3) its job is to simply enrich the records that the endpoint is returning, which is a common task for endpoints.

Reply:
Good points.

A couple of notes:
- It would be my preference to use async/await here ideally, just because its a bit less complex to read/understand (less nesting).
- More optional, but we could move the inner chunk of code into its own function within this file, that constructs and returns the supplementaryData object, just as a way to make the endpoint a bit smaller/easier to read (more modular), and the supplementary data code documentable and testable in isolation.

Reply:
Refactored to use `async`